### ISSUE TITLE
object(NaN, {foo: 42})(0.5) throws TypeError

### ISSUE THREAD

**Issue by mbostock** [d3/d3-interpolate#10]
The starting value should be treated as an empty object if it is not an object. So, instead of:

```
> d3_interpolate.value(NaN, {foo: 42})(0.5)
TypeError: Cannot use 'in' operator to search for 'foo' in NaN
    at object (/Users/mbostock/Development/d3-interpolate/build/d3-interpolate.js:44:18)
```

The input `NaN` should be treated as `{}`:

```
> d3_interpolate.value({}, {foo: 42})(0.5)
{foo: 42}
```

Related mbostock/d3#2688.

**Comment by krukon** [comment]
https://github.com/d3/d3-interpolate/pull/13 solves this problem.

**Comment by mbostock** [comment]
Fixed by #13.
